src/test/java/com/hivemq/codec/decoder/MQTTMessageDecoderTest.java

```diff
@@ -174,4 +176,149 @@ public void test_second_connect_received() {
+    public void test_connect_mqtt5_packet_size_too_large() {
+        final byte[] mqtt5Connect = {
+                // fixed header
+                //   type, reserved
+                0b0001_0000,
+                // remaining length
+                17,
+                // variable header
+                //   protocol name
+                0, 4, 'M', 'Q', 'T', 'T',
+                //   protocol version
+                5,
+                //   connect flags
+                (byte) 0b0000_0000,
+                //   keep alive
+                0, 0,
+                //   properties
+                0,
+                // payload
+                //   client identifier
+                0, 4, 't', 'e', 's', 't'};
+
+        test_connect_packet_size_too_large(mqtt5Connect);
+    }
+
+    @Test
+    public void test_connect_mqtt3_1_1_packet_size_too_large() {
+        final byte[] mqtt311Connect = {
+                // fixed header
+                //   type, reserved
+                0b0001_0000,
+                // remaining length
+                17,
+                // variable header
+                //   protocol name
+                0, 4, 'M', 'Q', 'T', 'T',
+                //   protocol version
+                5,
+                //   connect flags
+                (byte) 0b0000_0000,
+                //   keep alive
+                0, 0,
+                //   properties
+                0,
+                // payload
+                //   client identifier
+                0, 4, 't', 'e', 's', 't'};
+
+        test_connect_packet_size_too_large(mqtt311Connect);
+    }
+
+    @Test
+    public void test_connect_mqtt3_1_packet_size_too_large() {
+        final byte[] mqtt31Connect = {
+                // fixed header
+                //   type, reserved
+                0b0001_0000,
+                // remaining length
+                17,
+                // variable header
+                //   protocol name
+                0, 6, 'M', 'Q', 'T', 'T',
+                //   protocol version
+                3, 1,
+                //   connect flags
+                (byte) 0b0000_0000,
+                //   keep alive
+                0, 0,
+                //   properties
+                0,
+                // payload
+                //   client identifier
+                0, 4, 't', 'e', 's', 't'};
+
+        test_connect_packet_size_too_large(mqtt31Connect);
+    }
+
+
+    @Test
+    public void test_publish_mqtt5_packet_size_too_large() {
+        test_publish_packet_size_too_large(ProtocolVersion.MQTTv5);
+    }
+
+    @Test
+    public void test_publish_mqtt3_1_1_packet_size_too_large() {
+        test_publish_packet_size_too_large(ProtocolVersion.MQTTv3_1_1);
+    }
+
+    @Test
+    public void test_publish_mqtt3_1_packet_size_too_large() {
+        test_publish_packet_size_too_large(ProtocolVersion.MQTTv3_1);
+    }
+
+    private void test_publish_packet_size_too_large(final @NotNull ProtocolVersion protocolVersion) {
```

Comment:
that's a method, you can name it in normal java-way.
```suggestion
    private void testPublishPacketSizeTooLarge(final @NotNull ProtocolVersion protocolVersion) {
```

Reply:
test have the same name than in enterprise, those are legacy test, I rather keep them as they are just to be similar to the old ones